packages/mco/components/mco-dashboard/data-policy/parsers/applicationset-parser.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import { APPLICATION_TYPE, DRPC_STATUS } from '@odf/mco/constants';
+import {
+  DisasterRecoveryResourceKind,
+  useArgoApplicationSetResourceWatch,
+} from '@odf/mco/hooks';
+import {
+  ACMManagedClusterKind,
+  DRClusterAppsMap,
+  DRClusterKind,
+} from '@odf/mco/types';
+import {
+  findDRType,
+  findDeploymentClusters,
+  getProtectedPVCsFromDRPC,
+  getRemoteNamespaceFromAppSet,
+} from '@odf/mco/utils';
+import { getName, getNamespace } from '@odf/shared/selectors';
+import { WatchK8sResultsObject } from '@openshift-console/dynamic-plugin-sdk';
+import * as _ from 'lodash-es';
+
+export const useApplicationSetParser: UseApplicationSetParser = (
+  drResources,
+  managedClusters,
+  drLoaded,
+  drLoadError
+) => {
+  const [argoApplicationSetResources, loaded, loadError] =
+    useArgoApplicationSetResourceWatch({
+      overrides: { managedClusters: managedClusters },
+      drResources: {
+        data: drResources,
+        loaded: drLoaded,
+        loadError: drLoadError,
+      },
+    });
+
+  const drClusters: DRClusterKind[] = drResources?.drClusters;
+  const formattedArgoAppSetResources =
+    argoApplicationSetResources?.formattedResources;
+
+  const drClusterAppsMap: DRClusterAppsMap = React.useMemo(() => {
+    if (loaded && !loadError) {
+      const managedClusterList: ACMManagedClusterKind[] = Array.isArray(
+        managedClusters.data
```

Comment:
```suggestion
        managedClusters
```
as per the suggestion https://github.com/red-hat-storage/odf-console/pull/1114/files#r1426527449, managedClusters is ACMManagedCluster[] type